hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsConfigKeys.java

```diff
@@ -84,21 +84,29 @@
   public static final String HDDS_SCM_SAFEMODE_PIPELINE_AVAILABILITY_CHECK =
       "hdds.scm.safemode.pipeline-availability.check";
   public static final boolean
-      HDDS_SCM_SAFEMODE_PIPELINE_AVAILABILITY_CHECK_DEFAULT = false;
+      HDDS_SCM_SAFEMODE_PIPELINE_AVAILABILITY_CHECK_DEFAULT = true;
+
+  public static final String HDDS_SCM_SAFEMODE_PIPELINE_CREATION =
+      "hdds.scm.safemode.pipeline.creation";
+  public static final boolean
+      HDDS_SCM_SAFEMODE_PIPELINE_CREATION_DEFAULT = true;
 
   // % of containers which should have at least one reported replica
   // before SCM comes out of safe mode.
   public static final String HDDS_SCM_SAFEMODE_THRESHOLD_PCT =
       "hdds.scm.safemode.threshold.pct";
   public static final double HDDS_SCM_SAFEMODE_THRESHOLD_PCT_DEFAULT = 0.99;
 
-
   // percentage of healthy pipelines, where all 3 datanodes are reported in the
   // pipeline.
   public static final String HDDS_SCM_SAFEMODE_HEALTHY_PIPELINE_THRESHOLD_PCT =
       "hdds.scm.safemode.healthy.pipelie.pct";
   public static final double
       HDDS_SCM_SAFEMODE_HEALTHY_PIPELINE_THRESHOLD_PCT_DEFAULT = 0.10;
```

Comment:
Can we reuse HDDS_SCM_SAFEMODE_HEALTHY_PIPELINE_THRESHOLD_PCT_DEFAULT or consolidate the criteria?

Reply:
HDDS_SCM_SAFEMODE_THRESHOLD_PCT controls the open container percentage which has at least one replica reported,  to exit safe mode,  default value is 0.99. 
HDDS_SCM_SAFEMODE_HEALTHY_PIPELINE_THRESHOLD_PCT controls the percentage of healthy pipeline(with all datanodes reported), default value is 0.1.
I'm not sure why open container only consider one replica reported is enough.  From my understanding, open container with only one replica is not ready for use.  Maybe we should think about these criteria first.